File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        
        int Xor=0,n=nums.size();
        vector<int>res;
        for(int i:nums)Xor^=i;
        res.push_back(Xor^=((1<<maximumBit)-1));
        for(int i=1;i<n;i++){
            res.push_back(Xor^=nums[n-i]);
        }
        return res;
        
    }
};

File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int[] res=new int[nums.length];
        int Xor=0,n=nums.length;
        for(int i:nums)Xor^=i;
        res[0]=Xor^=(1<<maximumBit)-1;
        for(int i=1;i<n;i++){
            res[i]=Xor^=nums[n-i];
        }
        return res;
    }
}